float64[16] covariance
float64[3] innovation